src/sparseml/pytorch/recipe_template/main.py

```diff
@@ -0,0 +1,351 @@
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import warnings
+from collections import defaultdict
+from pathlib import Path
+from typing import Any, Dict, List, Optional, Tuple, Union
+
+import torch
+from torch.nn import Module
+
+from sparseml.pytorch.sparsification import (
+    ACDCPruningModifier,
+    ConstantPruningModifier,
+    EpochRangeModifier,
+    GlobalMagnitudePruningModifier,
+    LearningRateFunctionModifier,
+    MagnitudePruningModifier,
+    MFACPruningModifier,
+    MovementPruningModifier,
+    OBSPruningModifier,
+    QuantizationModifier,
+)
+from sparseml.pytorch.utils import get_prunable_layers, get_quantizable_layers
+from sparseml.sparsification import ModifierYAMLBuilder, RecipeYAMLBuilder
+
+
+__all__ = [
+    "recipe_template",
+]
+
+
+def recipe_template(
+    pruning: str = "false",
+    quantization: bool = False,
+    lr_func: str = "linear",
+    mask_type: str = "unstructured",
+    global_sparsity: bool = False,
+    target: Optional[str] = None,
+    model: Union[str, Module, None] = None,
+    file_name: str = "recipe.md",
+):
+    """
+    Returns a valid yaml or md recipe based on specified arguments
+
+    :pruning: the pruning algorithm to use in the recipe, can be any of the following,
+        `true` (represents Magnitude/Global-Magnitude pruning according to
+        global_sparsity), `false` (No pruning), `acdc`, `mfac`, `movement`, `obs` or
+        `constant`. Defaults to `false`
+    :quantization: True if quantization needs to be applied else False. Defaults to
+        False
+    :lr_func: the learning rate schedule function. Defaults to `linear`
+    :mask_type: the mask_type to use for pruning. Defaults to `unstructured`
+    :global_sparsity: if set to True then apply sparsity globally, defaults to
+        False
+    :target: the target hardware, can be set to `vnni` or `tensorrt`. Defaults to
+        None
+    :model: an instantiated PyTorch Module, or the local path to a torch.jit loadable
+        *.pt file, if supplied then the recipe is built according to this architecture
+    :file_name: the filename to save this recipe to. Defaults to `recipe.md`, if the
+        file extension is not markdown, then a yaml file is written.
+    """
+
+    recipe = _build_recipe(
+        global_sparsity=global_sparsity,
+        lr_func=lr_func,
+        mask_type=mask_type,
+        model=model,
+        pruning=pruning,
+        quantization=quantization,
+        target=target,
+        convert_to_md=Path(file_name).suffix == ".md",
+    )
+    _write_recipe_to_file(file_name=file_name, recipe=recipe)
+    return recipe
+
+
+def _build_recipe(
+    pruning: str = "false",
```

Comment:
Is pruning a bool as well? If actually a str, recommend changing to `Optional[str]=None` here

Reply:
removed the function as requested

Reply:
Why?

Reply:
Accurately representing the types a function supports makes it easier for users to use it and also when we go back later to look at this function. Seeing `:str = "false"` makes me very confused about what I can actually pass in here. `Union[bool, str]` could be good too, depending on what the types we accept actually are

Reply:
I accept this, am making this change.